slave/slave.go

```diff
@@ -8,17 +8,94 @@ import (
 	"github.com/aaronang/cong-the-ripper/lib"
 )
 
+var slaveInstance Slave
+
+type Slave struct {
+	heartbeat   lib.Heartbeat
+	successChan chan CrackerSuccess
+	failChan    chan CrackerFail
+	addTaskChan chan lib.Task
+}
+
+func Init(instanceId string) *Slave {
+	heartbeat := lib.Heartbeat{
+		SlaveId: instanceId,
+	}
+
+	slaveInstance = Slave{
+		heartbeat:   heartbeat,
+		successChan: make(chan CrackerSuccess),
+		failChan:    make(chan CrackerFail),
+		addTaskChan: make(chan lib.Task),
+	}
+	return &slaveInstance
+}
+
+func (s *Slave) Run() {
+	http.HandleFunc(lib.TasksCreatePath, taskHandler)
+	go http.ListenAndServe(lib.Port, nil)
+
+	for {
+		select {
+		case task := <-s.addTaskChan:
+			s.addTask(task)
+			go s.Execute(task)
+
+		case msg := <-s.successChan:
+			s.password_found(msg.taskID, msg.password)
+
+		case msg := <-s.failChan:
+			s.password_not_found(msg.taskID)
+		}
+	}
+}
+
 func taskHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
The `taskHandler` is the parameter for the `http.HandleFunc(lib.TasksCreatePath, taskHandler)` method, so there is no need to give it a pointer receiver.